llvm/lib/Analysis/TargetLibraryInfo.cpp

```diff
@@ -1337,6 +1337,8 @@ static const VecDesc VecFuncs_ArmPL[] = {
 #define TLI_DEFINE_ARMPL_VECFUNCS
 #define TLI_DEFINE_VECFUNC(SCAL, VEC, VF, MASK, VABI_PREFIX)                   \
   {SCAL, VEC, VF, MASK, VABI_PREFIX},
+#define TLI_DEFINE_VECFUNC_CC(SCAL, VEC, VF, MASK, VABI_PREFIX, CONV)          \
+  {SCAL, VEC, VF, MASK, VABI_PREFIX, CONV},
```

Comment:
Is a new macro required? It looks like TLI_DEFINE_VECFUNC is already library specific with some having a mask parameter and some not so I suspect you can just add the calling convention to the variant used by the one library you care about.

Whilst this kind of breaks the intent of this style of include, that boat looks to have sailed with the aforementioned mask change.

Reply:
I was trying to avoid adding a 6th parameter to the existing `TLI_DEFINE_VECFUNC` to reduce the size of the patch because for every other library function the 6th parameter will have to be `std::nullopt`, leading to an enormous patch. If you think it's better this way I'm happy to do it though.

Reply:
Oh I think I see what you mean now. You mean only change the definition of `TLI_DEFINE_VECFUNC` for the ARMPL variants to take a 6th argument. I can see if this is possible.